frontend/src/metabase/visualizer/utils/split-series.ts

```diff
@@ -4,34 +4,29 @@ import { isNotNull } from "metabase/lib/types";
 import { isCartesianChart } from "metabase/visualizations";
 import type {
   RawSeries,
-  VisualizationSettings,
   VisualizerColumnValueSource,
 } from "metabase-types/api";
 
 import { isDataSourceNameRef } from "./data-source";
 
 export function shouldSplitVisualizerSeries(
   columnValuesMapping: Record<string, VisualizerColumnValueSource[]>,
-  settings: VisualizationSettings,
 ) {
-  const dimensions = settings["graph.dimensions"] ?? [];
-  const dimensionDataSources = _.uniq(
-    dimensions
-      .map((columnName) => {
-        const mapping = columnValuesMapping[columnName];
-        if (!mapping || isDataSourceNameRef(mapping[0])) {
-          return;
-        }
-        return mapping[0]?.sourceId;
-      })
+  const mappings = Object.values(columnValuesMapping).flat();
+  const uniqueDataSources = _.uniq(
+    mappings
+      .map((mapping) =>
+        !isDataSourceNameRef(mapping) ? mapping.sourceId : null,
+      )
       .filter(isNotNull),
   );
-  return dimensionDataSources.length > 1;
+  return uniqueDataSources.length > 1;
 }
 
 export function splitVisualizerSeries(
```

Comment:
This function contains some pretty complex logic, are there unit tests for it?

Reply:
Not yet, but it's a good call out, I'll add some coverage in a followup PR 👍